src/overlays/actors/ovl_Bg_Dy_Yoseizo/z_bg_dy_yoseizo.c

```diff
@@ -38,56 +37,723 @@ ActorInit Bg_Dy_Yoseizo_InitVars = {
+                     Animation_GetLastFrame(sAnimations[GREATFAIRY_ANIM_SPIN_LAY_DOWN]), ANIMMODE_ONCE, 0.0f);
+    this->actionFunc = func_80A0AE1C;
+    Actor_PlaySfx(&this->actor, NA_SE_VO_FR_LAUGH_0);
+    Actor_PlaySfx(&this->actor, NA_SE_EV_GREAT_FAIRY_VANISH);
+    this->unk2F8 = 0;
+    this->actor.velocity.y = 0.0f;
+    this->unk2F0 = 0.0f;
+    this->unk2F4 = 0.0f;
+    this->actor.shape.yOffset = 0.0f;
+}
+
+void func_80A0B078(BgDyYoseizo* this, PlayState* play) {
+    BgDyYoseizo_Bob(this, play);
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_103) &&
+        (play->csCtx.actorCues[Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_103)]->id == 7)) {
+        Animation_Change(&this->skelAnime, sAnimations[GREATFAIRY_ANIM_LAYING_DOWN], 1.0f, 0.0f,
+                         Animation_GetLastFrame(sAnimations[GREATFAIRY_ANIM_LAYING_DOWN]), ANIMMODE_LOOP, 0.0f);
+        this->actionFunc = func_80A0B184;
+    } else if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_103) &&
+               (play->csCtx.actorCues[Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_103)]->id == 6)) {
+        func_80A0AFDC(this);
+    }
+}
+
+void func_80A0B184(BgDyYoseizo* this, PlayState* play) {
+    BgDyYoseizo_Bob(this, play);
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_103) &&
+        (play->csCtx.actorCues[Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_103)]->id == 8)) {
+        Animation_Change(&this->skelAnime, sAnimations[GREATFAIRY_ANIM_SHOWING_ITEM], 1.0f, 0.0f,
+                         Animation_GetLastFrame(sAnimations[GREATFAIRY_ANIM_SHOWING_ITEM]), ANIMMODE_LOOP, 0.0f);
+        this->actionFunc = func_80A0B078;
+    } else if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_103) &&
+               (play->csCtx.actorCues[Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_103)]->id == 6)) {
+        func_80A0AFDC(this);
+    }
+}
+
+void func_80A0B290(BgDyYoseizo* this, PlayState* play) {
+    BgDyYoseizo_Bob(this, play);
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_103) &&
+        (play->csCtx.actorCues[Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_103)]->id == 7)) {
+        Animation_Change(&this->skelAnime, sAnimations[GREATFAIRY_ANIM_LAYING_DOWN], 1.0f, 0.0f,
+                         Animation_GetLastFrame(sAnimations[GREATFAIRY_ANIM_LAYING_DOWN]), ANIMMODE_LOOP, -10.0f);
+        this->actionFunc = func_80A0B184;
+        this->mouthIndex = 0;
+    }
+}
+
+void func_80A0B35C(BgDyYoseizo* this, PlayState* play) {
+    BgDyYoseizo_Bob(this, play);
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->timer == 60) {
+        if (!Flags_GetSwitch(play, GREAT_FAIRY_GET_SWITCHFLAG(&this->actor))) {
+            switch (GREAT_FAIRY_GET_TYPE(&this->actor)) {
+                case GREAT_FAIRY_TYPE_MAGIC:
+                    if (gSaveContext.save.saveInfo.playerData.isMagicAcquired != true) {
```

Comment:
```suggestion
                    if (gSaveContext.save.saveInfo.playerData.isMagicAcquired) {
```
since it's a boolean, can't remember what the overall consensus was on writing these checks.

Reply:
This is one of those weird fake booleans that end up checking `!= 1` it seems, since devs were loose with this

Reply:
The comparison to `true` is sadly required

I wonder if instead of a boolean value this is more of a state that just happens to be 0 and 1. Something like `MAGIC_STATE_0` and `MAGIC_STATE_1` idk, either way doesn't matter too much